### PR TITLE
Adjust workflows

### PR DESCRIPTION
**Pull request by micorix**
* Run linter within docker container in github actions
* Remove separate linting workflow
* Remove dockerfile for postgres. Pin postgres directly in docker-compose file
* Add concurrency and timeout to test building workflow
* Run test building workflow only on PRs

* Format dockerfile
* Add .idea to gitignore

### CODE REVIEW

**Comment by micorix** [comment]
@Anakin100100 PR is ready but tests fail bc of the same error I encountered locally.

**Comment by Anakin100100** [comment]
This is not really the same error. Here several gems are not found even though they should be present and your issue was with the wrappers for pg gem executable. These are totally different and it was solved by the pg gem version upgrade. 

I don't see any  obvious changes that could trigger this error. @wojtodzio maybe you can see sth? 

Because the changes you were introducing on this branch were causing major errors from the begginig we don't know when this bug was introduced. Now that the changes that we want to apply are well defined we should backtrack to the working state on the data_filtering branch and start introducing the changes from smallest to largest and wait until this one pops up.  

Maybe the error is related to layer caching somehow? Missing gems would suggest that because it's literraly impossible only for them not to be installed. Probably more gems are missing but these have caused fatal errors first before the others could.

**Comment by wojtodzio** [comment]
I'm not sure what's causing the issue, but build cache sounds like a reasonable guess. Either that, or there might be something wrong with paths, but the Dockerfile seems fine to me.

File: .ci/entrypoint.tests.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/ash
```
Comment (wojtodzio) [inline .ci/entrypoint.tests.sh:1]:
```suggestion
#!/usr/bin/env bash
```

**Comment by Anakin100100** [comment]
@micorix there seems to be no obvious solution, could we merge that pull request and work on it later? Finding one is probably going to be quite lengthy because we are dealing with a non-obvious bug that has something to do with advanced internarls of software we don't know very well. The config on data_filtering has some stylistic flaws but does not have a bug that we don't know how to deal with and gets the job done. It catches the build errors from docker and ensures that our code runs properly on a container. Being pretty is kinda secondary here especially considering that we are pressured to release this soon.  @mlazowik could you take a look on this? This would fall more into your area of expertise.

**Comment by micorix** [comment]
1. In my opinion master branch should always work and be ready to deploy. I would even state that fixing this is somehow important cause we'll use Docker to deploy so it wouldn't work on prod anyway

2. Besides what's the point of running tests in virtualized environment which is supposed to work for everyone if we decide to just ignore test/build errors? 

3. I'm not sure whether this is cache issue but we can remove `cache-from` and see if it works.

**Comment by mlazowik** [comment]
Looks like what fails here is executing the tests, not building itself, so there is some possibility that prod will work.

While the project is in heavy development initial stages, especially if CI never really worked, I'd say it's fine to merge.

Fixing CI should be hight priority, no prod release should be done without CI being fixed. Once CI is properly in place it should not be ignored.

**Comment by mlazowik** [comment]
Even if disabling cache fixes the problem it'd still mean that the problem is with the Dockerfile(s). For example not all files that impact dependency resolution are copied before dependency installation or something similar.

**Comment by Anakin100100** [comment]
The cicd on data_filtering is working. We started introducing lots of changes that shpoldn't impact the functionality to a high degree but we messed something up and it's not working. @micorix Where is the issue with merging data_filtering? The setup on data_filtering does what it is supposed to do and works properly. There are no indicators that it would break in production env. If it doesn't work there (very unlikely) it will give us more info to pinpoint the issue. Imo this bug is related to cicd setup and not rails code or the dockerfile.

**Comment by mlazowik** [comment]
Has anyone tried running the tests locally in the same way they are run in CI?

**Comment by mlazowik** [comment]
Either way if there's no working test deployment yet of the rails backend IMO it's ok to merge and figure it out later.

**Comment by mlazowik** [comment]
> If everything is fine with the code and the dockerfile then the bug is in the cicd config.
Not necessarily, docker image builds are usually not deterministic

**Comment by Anakin100100** [comment]
I'm kinda out of ideas when it comes to problems with the dockerfile and the code. @micorix @wojtodzio @mlazowik do you have any new ideas so far?

**Comment by mlazowik** [comment]
> For example not all files that impact dependency resolution are copied before dependency installation or something similar.

Hehe

**Comment by micorix** [comment]
:tada: Checks pass! 

`Gemfile.lock` file was not copied to the image so there were some problems with deps resolution. 

Marking this PR as ready for final review

**Review by mlazowik — APPROVED** [review]

**Review by Anakin100100 — APPROVED** [review]
Lgtm

**Pull request merged**